tests/api/test_user_resource.py

```diff
@@ -149,25 +143,12 @@ def test_user_deactivate(self, client, organization, token_generator):
             '/clients/{}'.format(associated_client_id), headers=headers)
         expect(response.json['response']['client_secret']).to(be(None))
 
-        # DELETE the data trusts and by extension delete the users and associated client
-        response = client.delete(
-            '/data_trusts/{}'.format(data_trust_id), headers=headers)
+        # Delete users and clients
+        response = client.delete(f'/clients/{associated_client_id}', headers=headers)
         expect(response.status_code).to(be(200))
-
-    def _post_data_trust(self, client, token_generator):
-        '''
-        Helper function that creates (and tests creating) a Data Trust entity.
-        '''
-        request_body = {
-            'data_trust_name': 'BrightHive Test Data Trust'
-        }
-        headers = {'content-type': 'application/json', 'authorization': f'bearer {token_generator.get_token(client)}'}
-        response = client.post(
-            '/data_trusts', data=json.dumps(request_body), headers=headers)
-        expect(response.status_code).to(be(201))
-        data_trust_id = response.json['response'][0]['id']
-
-        return data_trust_id
+        for user_id in added_user_ids:
+            response = client.delete(f"/users/{user_id}", headers=headers)
+            expect(response.status_code).to(be(200))
```

Comment:
I'd like to see `test_user_api` include a test to "Update the role of a user" (just can do that here: https://github.com/brighthive/authserver/pull/35/files#diff-4cb198d7da3a7f0c5588bb1071a57903R75). 

Assigning a new role to an existing user seems like a use case worth testing.